pkg/util/kube/utils_test.go

```diff
@@ -120,3 +124,81 @@ func TestEnsureNamespaceExistsAndIsReady(t *testing.T) {
 	}
 
 }
+
+type harness struct {
+	*test.APIServer
+
+	log logrus.FieldLogger
+}
+
+func newHarness(t *testing.T) *harness {
+	t.Helper()
+
+	return &harness{
+		APIServer: test.NewAPIServer(t),
+		log:       logrus.StandardLogger(),
+	}
+}
+
+// TestGetVolumeDirectoryWithPVCSuccess tests that the GetVolumeDirectory function
+// returns a volume's name or a volume's name plus '/mount' when a PVC is present.
+func TestGetVolumeDirectorySuccess(t *testing.T) {
+	tests := []struct {
+		name            string
+		pvc             *corev1.PersistentVolumeClaim
+		pv              *corev1.PersistentVolume
+		csiVolumeSource *corev1.CSIVolumeSource
+		want            string
+	}{
+		{
+			name: "Non-CSI volume with a PVC/PV returns the volume's name",
+			pvc:  test.NewPVC("ns-1", "my-pvc", test.WithVolumeName("a-pv")),
+			pv:   test.NewPV("a-pv"),
+			want: "a-pv",
+		},
+		{
+			name: "CSI volume with a PVC/PV appends '/mount' to the volume name",
+			pvc:  test.NewPVC("ns-1", "my-pvc", test.WithVolumeName("a-pv")),
+			pv:   test.NewPV("a-pv", test.WithCSI("csi.test.com", "provider-volume-id")),
+			want: "a-pv/mount",
+		},
+		{
+			name:            "CSI volume mounted without a PVC appends '/mount' to the volume name",
+			csiVolumeSource: &corev1.CSIVolumeSource{Driver: "csi.test.com"},
+			want:            "my-vol/mount",
+		},
+		{
+			name: "Non-CSI volume without a PVC returns the volume name",
+			want: "my-vol",
+		},
+	}
+
+	for _, tc := range tests {
+		h := newHarness(t)
+
+		pvcInformer := kubeinformers.NewSharedInformerFactoryWithOptions(h.KubeClient, 0, kubeinformers.WithNamespace("ns-1")).Core().V1().PersistentVolumeClaims()
+		pvInformer := kubeinformers.NewSharedInformerFactory(h.KubeClient, 0).Core().V1().PersistentVolumes()
+
+		vol := &corev1.Volume{Name: "my-vol"}
```

Comment:
I've generally preferred just defining things like this in the test cases using struct literals or helpers, rather than dynamically constructing them during  test execution -- although it can be a little more verbose, I think it improves clarity both in the test cases and in the body of the test. So L182,186,192-194,196 would go away and the test case struct would get a `pod` field that includes the volume you want.

Reply:
That's a good suggestion - I was trying to avoid attaching structs within the test case to each other, but I'm reviewing it and see that this should be an issue here.